src/tensor/problem_description.cpp

```diff
@@ -0,0 +1,67 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2023 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <miopen/tensor/problem_description.hpp>
+#include <miopen/names.hpp>
+
+namespace miopen {
+
+namespace tensor {
+
+NetworkConfig ProblemDescription::MakeNetworkConfig() const
```

Comment:
Several questions about PD (Problem Description) network config:
1. We have alphas and beta in PD  Should we add them in NetworkConfig ? Or probably we need remove alphas from PD at all  and add only beta to NetworkConfig?
2.  Should we add tensor sizes to NetworkConfig ?
@CAHEK7  your thoughts ?

Reply:
1 and 2: each problem descriptor should provide a unique network config, so sizes, strides, alphas, betas, etc. must be a part of the network config.

1: parameters from the problem description and from the invoke params should not duplicate each other.
problem description represents static problem parameters, may affect compilation and performance cannot be changed in runtime, while invoke params represents runtime parameters which can be changed for a particular problem (for example data pointers). In this particular case, we should check which parameter can be a pure runtime parameter (alphas is a good candidate, while beta=0 or beta!=0 may affect compilation) and move them outside of the problem description.